routers/api/v1/org/org.go

```diff
@@ -75,6 +75,7 @@ func Create(ctx *context.APIContext, form api.CreateOrgOption) {
 	// parameters:
 	// - name: organization
 	//   in: body
+	//   description: "'visibility' can be \"public\", \"limited\" or \"private\""
```

Comment:
You should use enum to provide the valid values

Reply:
E.g. https://github.com/go-gitea/gitea/blob/c385dcc26b0da7fc1c245a1da316551cb6585ad3/routers/api/v1/repo/pull.go#L48

Reply:
```suggestion
	//   enum: [public, limited, private]
	//   description: "'visibility' can be \"public\", \"limited\" or \"private\""
```

Reply:
Looks like you can only do this in a query, not the body.

Reply:
Sorry I misread the swagger here - I didn't realise this was referring to a CreateOrgOption. The enum needs to be in struct.

Here's a patch: (I'd have opened a PR to your PR but I don't have access to it.)

```patch
diff --git a/modules/structs/org.go b/modules/structs/org.go
index aa76b9120..a99197ffc 100644
--- a/modules/structs/org.go
+++ b/modules/structs/org.go
@@ -24,7 +24,8 @@ type CreateOrgOption struct {
        Description string `json:"description"`
        Website     string `json:"website"`
        Location    string `json:"location"`
-       Visibility  string `json:"visibility" binding:"In(,visible,limited,private)"`
+       // enum: public,limited,private
+       Visibility string `json:"visibility" binding:"In(,visible,limited,private)"`
 }
 
 // EditOrgOption options for editing an organization
@@ -33,5 +34,6 @@ type EditOrgOption struct {
        Description string `json:"description"`
        Website     string `json:"website"`
        Location    string `json:"location"`
-       Visibility  string `json:"visibility" binding:"In(,visible,limited,private)"`
+       // enum: public,limited,private
+       Visibility string `json:"visibility" binding:"In(,visible,limited,private)"`
 }
diff --git a/templates/swagger/v1_json.tmpl b/templates/swagger/v1_json.tmpl
index 846ddec79..e7a7e6c3a 100644
--- a/templates/swagger/v1_json.tmpl
+++ b/templates/swagger/v1_json.tmpl
@@ -7144,6 +7144,11 @@
         },
         "visibility": {
           "type": "string",
+          "enum": [
+            "public",
+            "limited",
+            "private"
+          ],
           "x-go-name": "Visibility"
         },
         "website": {
@@ -7658,6 +7663,11 @@
         },
         "visibility": {
           "type": "string",
+          "enum": [
+            "public",
+            "limited",
+            "private"
+          ],
           "x-go-name": "Visibility"
         },
         "website": {
```

Or pull from https://github.com/zeripath/gitea/tree/fix-7023-api-org-visibility

Reply:
Ok. Merged in. Thanks!